### PR TITLE
New design for agg builder sprinkles

### PR DESCRIPTION
**Pull request by mmarcon**
With this PR we tweak the design of the stage builder so the info sprinkle is moved next to the stage name in the preview area and contains a tooltip that explains in short what the stage does.

~Unit tests are not updated yet.~

image (5)

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -70,51 +70,51 @@ export const TOOLTIP_OPEN_SAVED_PIPELINES = 'Open saved Pipelines';
 export const STAGE_SPRINKLE_MAPPINGS = {
   $addFields: {
     link: 'https://docs.mongodb.com/manual/reference/operator/aggregation/addFields/#pipe._S_addFields',
-    tooltip: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,<br />sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.<br />Ut enim ad minim veniam, quis nostrud exercitation<br />ullamco laboris nisi ut aliquip ex ea commodo consequat.'
+    tooltip: 'Adds new field(s) to a document with a computed value, or reassigns an existing field(s) with a computed value.'
```
Comment (sampoonachot) [inline src/constants.js]: Screen Shot 2019-08-29 at 9 02 37 AM

@mmarcon I added the 1-liners and fixed the width of the tooltip so we no longer need to use `<br />`.

File: src/constants.js [hunk 2]
```diff
@@ -70,51 +70,51 @@ export const TOOLTIP_OPEN_SAVED_PIPELINES = 'Open saved Pipelines';
 export const STAGE_SPRINKLE_MAPPINGS = {
   $addFields: {
     link: 'https://docs.mongodb.com/manual/reference/operator/aggregation/addFields/#pipe._S_addFields',
```
Comment (sampoonachot) [inline src/constants.js]: Instead of deeplinking the info sprinkle, can we link the text `$stageName`?

**Comment by mmarcon** [comment]
@sampoonachot Now deeplinking from the stage name. I also left the link there in the sprinkle I think it's not a bad idea to have it in both places.

image

File: src/components/stage-preview-toolbar/stage-preview-toolbar.jsx [hunk 3]
```diff
@@ -49,7 +79,14 @@ class StagePreviewToolbar extends PureComponent {
         if (this.props.stageOperator === OUT && this.props.isValid) {
           return `Documents will be saved to the collection: ${decomment(this.props.stageValue)}`;
         }
-        return `Output after ${this.props.stageOperator} stage (Sample of ${this.props.count} ${this.getWord()})`;
+        const stageInfo = STAGE_SPRINKLE_MAPPINGS[this.props.stageOperator];
+        return (
+          <div>
+            <span>Output after <span onClick={this.props.openLink.bind(this, stageInfo.link)} className={classnames(styles['stage-preview-toolbar-link'])}>{this.props.stageOperator}</span> stage</span>
```
Comment (lrlna) [inline src/components/stage-preview-toolbar/stage-preview-toolbar.jsx]: what do you think about multilining this (i know it's only a `span` butttt) for future readability?

File: src/components/stage-preview-toolbar/stage-preview-toolbar.less [hunk 4]
```diff
@@ -1,8 +1,39 @@
...
+    color: @blue3;
+  }
+
+  :global {
+    i.info-sprinkle {
+      display: inline-block;
+      font: normal normal normal 14px/1 FontAwesome;
+      font-size: inherit;
+      text-rendering: auto;
+      -webkit-font-smoothing: antialiased;
```
Comment (lrlna) [inline src/components/stage-preview-toolbar/stage-preview-toolbar.less]: I _feel_ like `-webkit` and `-mox-osx` are for safari and firefox respectively (might be wrong tho idk). Would we be able to use just [`font-smoothing`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-smooth) for this, do you think?

Comment (mmarcon) [reply src/components/stage-preview-toolbar/stage-preview-toolbar.less]: I think the prefix is needed (`-webkit` also applies to Chrome). `font-smooth` is non-standard at this point: https://caniuse.com/#feat=font-smooth

Comment (lrlna) [reply src/components/stage-preview-toolbar/stage-preview-toolbar.less]: ok! sounds good!

**Comment by sampoonachot** [comment]
lgtm

**Pull request merged**